File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/RecommendationScenario.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// RecommendationScenario Declaration
//==============================================================
type RecommendationScenario struct {
    gorm.Model
     Name                                    string
    Objective                                    string
     Models           []Model_ `gorm:"foreignKey:ModelsFromRecommendationScenarioId"`
     Datasets           []DataSet `gorm:"foreignKey:DatasetsFromRecommendationScenarioId"`
     Experiments           []Experiment `gorm:"foreignKey:ExperimentsFromRecommendationScenarioId"`
     Alerts           []Alert `gorm:"foreignKey:AlertsFromRecommendationScenarioId"`
    RecommendationType                      RecommendationType

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/analytics-on-rails/analyticsonrails/app/models/recommendationscenario.rb
class RecommendationScenario < ApplicationRecord
  enum RecommendationType: [:Personalized, :Trending, :SimilarItems, :FrequentlyBoughtTogether, :ContentBased]


  has_many :Models, class_name: 'Model'
  has_many :Datasets, class_name: 'DataSet'
  has_many :Experiments, class_name: 'Experiment'
  has_many :Alerts, class_name: 'Alert'

end
